apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/index.ts

```diff
@@ -48,18 +48,28 @@ export const validateiOSAttestation = (
     E.chainW(validate(iOsAttestation, "iOS attestation format is invalid")),
     TE.fromEither,
     TE.chain((decodedAttestation) =>
-      TE.tryCatch(
-        () =>
-          verifyAttestation({
-            decodedAttestation,
-            challenge,
-            keyId,
-            bundleIdentifier,
-            teamIdentifier,
-            allowDevelopmentEnvironment,
-            appleRootCertificate,
-          }),
-        E.toError
+      pipe(
+        TE.tryCatch(
+          () =>
+            verifyAttestation({
+              decodedAttestation,
+              challenge,
+              keyId,
+              bundleIdentifier,
+              teamIdentifier,
+              allowDevelopmentEnvironment,
+              appleRootCertificate,
+            }),
+          E.toError
+        ),
+        TE.chainW(({ hardwareKey }) =>
+          pipe(
+            hardwareKey,
+            validate(JwkPublicKey, "Invalid JWK Public Key"),
+            E.map((hardwareKey) => ({ hardwareKey })),
+            TE.fromEither
```

Comment:
I would make this small change to avoid that if new output parameters are added to the verifyAttestation, there isn't an explicit warning reminding you to add them here too. 

```suggestion
        TE.chainW((result) =>
          pipe(
            result.hardwareKey,
            validate(JwkPublicKey, "Invalid JWK Public Key"),
            E.map((hardwareKey) => ({ ...result, hardwareKey })),
            TE.fromEither
```